qemu/tests/blockdev_commit_install.py

```diff
@@ -0,0 +1,57 @@
+import time
+import logging
+import random
+
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.tests import unattended_install
+
+from provider.blockdev_commit_base import BlockDevCommitTest
+
+
+class BlockdevCommitInstall(BlockDevCommitTest):
+    def configure_system_disk(self, tag):
+        pass
+
+
+def run(test, params, env):
+    """
+    Block commit base Test
+
+    1. Install guest
+    2. create 4 snapshots during guest installation
+    3. commit snapshot 3 to base
+    4. installation can be finished after commit
+    """
+    def tag_for_install(vm, tag):
+        if vm.serial_console:
+            serial_output = vm.serial_console.get_output()
+            if tag in serial_output:
+                return True
+        logging.info("vm has not started yet")
+        return False
+
+    block_test = BlockdevCommitInstall(test, params, env)
+    args = (test, params, env)
+    bg = utils_test.BackgroundTest(unattended_install.run, args)
+    bg.start()
+    if bg.is_alive():
+        tag = params.get("tag_for_install_start", "Starting Login Service")
+        if utils_misc.wait_for(lambda: tag_for_install(block_test.main_vm, tag), 240, 10, 5):
+            logging.info("sleep random time before do snapshots")
+            time.sleep(random.randint(10, 120))
+            block_test.pre_test()
+            try:
+                block_test.commit_snapshots()
```

Comment:
One concern here, from log:

12:21:23 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'blockdev-snapshot'
12:21:23 DEBUG| Send command: {'execute': 'blockdev-snapshot', 'arguments': {'overlay': 'drive_sn1', 'node': 'drive_image1'}, 'id': 'GLmldN8y'}
12:21:24 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'blockdev-snapshot'
12:21:24 DEBUG| Send command: {'execute': 'blockdev-snapshot', 'arguments': {'overlay': 'drive_sn2', 'node': 'drive_sn1'}, 'id': 'mf0xIa3e'}
12:21:24 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'blockdev-snapshot'
12:21:24 DEBUG| Send command: {'execute': 'blockdev-snapshot', 'arguments': {'overlay': 'drive_sn3', 'node': 'drive_sn2'}, 'id': 'f8efICz8'}
12:21:24 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'blockdev-snapshot'
12:21:24 DEBUG| Send command: {'execute': 'blockdev-snapshot', 'arguments': {'overlay': 'drive_sn4', 'node': 'drive_sn3'}, 'id': 'aF6ZxCUC'}

Please check the time that, the interval of snapshots is too small to write enough data into image1, so what happened? please check the log below:

12:21:24 DEBUG| (monitor avocado-vt-vm1.qmpmonitor1) Sending command 'block-commit'
12:21:26 DEBUG| [stdout] lzc: event: JOB_STATUS_CHANGE: BLOCK_JOB_COMPLETED
12:21:26 DEBUG| [stdout] lzc: event: BLOCK_JOB_COMPLETED: BLOCK_JOB_COMPLETED

The 'lzc' line is the debug info I made, you can see that the commit job finished in 2s, I am afraid this might not be so good for our testing, for the data written between snapshots was so little that it took very short time(2s) to finish the commit.

Reply:
Although the interval for snapshots set to 90s, it just takes 6s for commit job to finish.
So, will keep the original design, thanks for your suggestion.